idaes/core/util/tables.py

```diff
@@ -235,10 +235,18 @@ class VariableTypes:
 
     for key, sb in stream_states.items():
         stream_attributes[key] = {}
-        if true_state:
-            disp_dict = sb.define_state_vars()
-        else:
-            disp_dict = sb.define_display_vars()
+        try:
+            if true_state:
+                disp_dict = sb.define_state_vars()
+            else:
+                disp_dict = sb.define_display_vars()
+        except AttributeError:  # if define_(display,state)_vars doesn't exist(!)
```

Comment:
Have we actually seen cases of this? In order for `define_state_vars` not to exist then someone must have done something wrong and maybe we should be catching this earlier.

Reply:
Yes, in Mayo's model

Reply:
However if you can suggest where "earlier" might be, I can try to catch it there

Reply:
So, I think this might need a bigger design discussion. The issue here is the the stream table methods were written only to consider things that were in "material streams", i.e. those related to state blocks in which case the `define_state_vars` method must exist.

What Mayo was doing was trying to add additional, non-state, data to the stream table which is not supported at the moment. I think a better solution to this might be to design an interface for adding this additional information rather than trying to shoehorn it into the existing interface. There were a few other catches to what Mayo was trying to do as well due to how state data is indexed versus things at the unit level that need to be resolved as well.